packages/core/database/src/schema/diff.ts

```diff
@@ -89,7 +89,11 @@ export default (db: Database) => {
   const diffIndexes = (oldIndex: Index, index: Index): IndexDiff => {
     const changes: string[] = [];
 
-    if (!_.isEqual(oldIndex.columns, index.columns)) {
+    // Sort columns before comparing
+    const sortedOldColumns = [...oldIndex.columns].sort();
+    const sortedNewColumns = [...index.columns].sort();
+
+    if (!_.isEqual(sortedOldColumns, sortedNewColumns)) {
```

Comment:
you could leave all of this to lodash using the _.xor if you don't want to bother with the spreading & sort